### PR TITLE
Feat: Onboard Census Bureau ACS Dataset

### PR DESCRIPTION
**Pull request by Naveen130**
## Description

Pipelines included:

- cbsa_2019_1yr
- cbsa_2020_1yr
- cbsa_2021_1yr
- cbsa_2019_5yr
- cbsa_2020_5yr
- cbsa_2021_5yr
- censustract_2019_5yr
- censustract_2020_5yr
- censustract_2021_5yr
- congressionaldistrict_2019_1yr
- congressionaldistrict_2020_1yr
- congressionaldistrict_2021_1yr
- congressionaldistrict_2019_5yr
- congressionaldistrict_2020_5yr
- congressionaldistrict_2021_5yr
- county_2019_1yr
- county_2020_1yr
- county_2021_1yr
- county_2019_5yr
- county_2020_5yr
- county_2021_5yr
- place_2019_1yr
- place_2020_1yr
- place_2021_1yr
- place_2019_5yr
- place_2020_5yr
- place_2021_5yr
- puma_2019_1yr
- puma_2020_1yr
- puma_2021_1yr
- puma_2019_5yr
- puma_2020_5yr
- puma_2021_5yr
- schooldistrictelementary_2016_1yr
- schooldistrictelementary_2017_1yr
- schooldistrictelementary_2018_1yr
- schooldistrictelementary_2019_1yr
- schooldistrictelementary_2020_1yr
- schooldistrictelementary_2021_1yr
- schooldistrictelementary_2016_5yr
- schooldistrictelementary_2017_5yr
- schooldistrictelementary_2018_5yr
- schooldistrictelementary_2019_5yr
- schooldistrictelementary_2020_5yr
- schooldistrictelementary_2021_5yr
- schooldistrictsecondary_2016_1yr
- schooldistrictsecondary_2017_1yr
- schooldistrictsecondary_2018_1yr
- schooldistrictsecondary_2019_1yr
- schooldistrictsecondary_2020_1yr
- schooldistrictsecondary_2021_1yr
- schooldistrictsecondary_2016_5yr
- schooldistrictsecondary_2017_5yr
- schooldistrictsecondary_2018_5yr
- schooldistrictsecondary_2019_5yr
- schooldistrictsecondary_2020_5yr
- schooldistrictsecondary_2021_5yr
- schooldistrictunified_2016_1yr
- schooldistrictunified_2017_1yr
- schooldistrictunified_2018_1yr
- schooldistrictunified_2019_1yr
- schooldistrictunified_2020_1yr
- schooldistrictunified_2021_1yr
- schooldistrictunified_2016_5yr
- schooldistrictunified_2017_5yr
- schooldistrictunified_2018_5yr
- schooldistrictunified_2019_5yr
- schooldistrictunified_2020_5yr
- schooldistrictunified_2021_5yr
- state_2019_1yr
- state_2020_1yr
- state_2021_1yr
- state_2019_5yr
- state_2020_5yr
- state_2021_5yr
- zcta_2019_5yr
- zcta_2020_5yr
- zcta_2021_5yr
## Checklist

Note: If an item applies to you, all of its sub-items must be fulfilled

- [x] **(Required)** This pull request is appropriately labeled
- [x] Please merge this pull request after it's approved
- [x] I'm adding or editing a dataset
  - [x] The [Google Cloud Datasets team](mailto:cloud-datasets-onboarding@google.com) is aware of the proposed dataset
  - [x] I put all my code inside  `datasets/<DATASET_NAME>` and nothing outside of that directory
- [x] I'm refactoring or cleaning up some code

### CODE REVIEW

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 1]
```diff
@@ -13,126 +13,480 @@
...
+    logging.info(f"Defining table schema... {bucket_name} ... {schema_filepath}")
+    schema = []
+    if not (schema_filepath):
+        schema_struct = schema_structure
+    else:
+        storage_client = storage.Client()
+        bucket = storage_client.get_bucket(bucket_name)
+        blob = bucket.blob(schema_filepath)
+        schema_struct = json.loads(blob.download_as_string(client=None))
+
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:415]: remove unnecessary blank lines within functions

**Comment by Naveen130** [comment]
Sure, will take care of the extra blank lines and comments.

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 2]
```diff
@@ -12,97 +12,432 @@
...
 
-    logging.info(
-        f"ACS {pipeline_name} process started at "
-        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
-    )
-
     logging.info("Creating 'files' folder")
     pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+    logging.info(f"{pipeline_name} process started")
+
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:56]: Here

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 3]
```diff
@@ -12,97 +12,432 @@
...
+    api_naming_convention: str,
     geography: str,
     report_level: str,
-    concat_col: typing.List[str],
+    concat_col_list: typing.List[str],
+    data_dtypes: str,
+    rename_mappings_list: dict,
+    input_csv_headers: typing.List[str],
+    output_csv_headers: typing.List[str],
 ) -> None:
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:52]: Here

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 4]
```diff
@@ -12,97 +12,432 @@
...
     elif report_level == "state_level":
         df = extract_data_and_convert_to_df_state_level(
-            group_id, state_code, year_report, api_naming_convention, source_url
+            group_id,
+            state_code,
+            year_report,
+            api_naming_convention,
+            source_url,
+            destination_table,
         )
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:120]: Here

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 5]
```diff
@@ -12,97 +12,432 @@
...
+                    rename_mappings_list=rename_mappings_list,
+                    concat_col_list=concat_col_list,
+                    input_csv_headers=input_csv_headers,
+                    output_csv_headers=output_csv_headers,
+                    group_id=group_id,
+                    state_code=state_code,
+                )
+                data = []
+                chunk_number += 1
+
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:208]: Here

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 6]
```diff
@@ -12,97 +12,432 @@
...
+                data_dtypes=data_dtypes,
+                geography=geography,
+                rename_mappings_list=rename_mappings_list,
+                concat_col_list=concat_col_list,
+                input_csv_headers=input_csv_headers,
+                output_csv_headers=output_csv_headers,
+                group_id=group_id,
+                state_code=state_code,
+            )
+
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:224]: Here

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 7]
```diff
@@ -12,97 +12,432 @@
...
+    dfcolumns = list(target_df.columns)
+    i = 0
+    while i < (len(output_csv_headers)):
+        if output_csv_headers[i] not in dfcolumns:
+            output_csv_headers.pop(i)
+            i -= 1
+        i += 1
+
```
Comment (nlarge-google) [inline datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py:232]: Here

**Review by nlarge-google — CHANGES_REQUESTED** [review]
Still some lines that were missed.

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py [hunk 8]
```diff
@@ -13,126 +13,480 @@
...
+    geography: str,
+    rename_mappings_list: dict,
+    concat_col_list: typing.List[str],
+    input_csv_headers: typing.List[str],
+    output_csv_headers: typing.List[str],
+    group_id: str,
+    state_code: str,
+) -> None:
+    df = pd.DataFrame(data, columns=input_headers)
+    # set_df_datatypes(df, data_dtypes)
```
[the remark this answers is not in the document]
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 1 above)
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 3 above)
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 4 above)
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 5 above)
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 6 above)
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

File: datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py (hunk 7 above)
Comment (Naveen130) [reply datasets/census_bureau_acs/pipelines/_images/run_csv_transform_kub/csv_transform.py]: Done

**Review by nlarge-google — APPROVED** [review]
Looks Great.

**Pull request merged**